File: python/024_bob/bob.py
def response(hey_bob):
    r = ["Whatever.", "Whoa, chill out!", "Sure.", "Calm down, I know what I'm doing!"]
    t = hey_bob.strip()
    if not t:
        return "Fine. Be that way!"
    s = 1 if t.isupper() else 0
    q = 2 if t[-1] == "?" else 0
    return r[s + q]

File: csharp/058_bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    private static readonly string[] Answers = {
        "Fine. Be that way!",
        "Whatever.",
        "Whoa, chill out!",
        "Sure.",
        "Calm down, I know what I'm doing!"
    };
    
    public static string Response(string statement)
    {
        statement = statement.Trim();
        bool isSomething = statement.Length > 0;
        bool isUpper = Regex.IsMatch(statement, "[A-Z]") && !Regex.IsMatch(statement, "[a-z]");
        bool isQuestion = statement.EndsWith("?");
        int index = Convert.ToInt32(isSomething) 
                  + Convert.ToInt32(isUpper) 
                  + (Convert.ToInt32(isQuestion) * 2);
        return Answers[index];
    }
}